### PR TITLE
feat: new workflow properties tab

### PR DESCRIPTION
**Pull request by AndrasEszes**
[CI-2762](https://bitrise.atlassian.net/browse/CI-2762)

[CI-2762]: https://bitrise.atlassian.net/browse/CI-2762?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: source/javascripts/components/WorkflowConfigPanel/WorkflowConfigPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  const form = useForm<FormValues>({
+    defaultValues: {
+      workflowId,
+      ...defaultValues,
+    },
+  });
+
+  const handleSubmit = form.handleSubmit((data, e) => {
+    e?.preventDefault();
+    e?.stopPropagation();
```
Comment (zoltanszabo-bitrise) [inline source/javascripts/components/WorkflowConfigPanel/WorkflowConfigPanel.tsx:25]: [Q]: Aren't these **called automatically** by `handleSubmit`?

**Review by zoltanszabo-bitrise — DISMISSED** [review]

File: source/javascripts/components/WorkflowConfigPanel/WorkflowConfigPanel.tsx (hunk 1 above)
Comment (AndrasEszes) [reply source/javascripts/components/WorkflowConfigPanel/WorkflowConfigPanel.tsx:25]: It seems, the preventDefault method is invoked: https://github.com/react-hook-form/react-hook-form/blob/master/src/logic/createFormControl.ts#L1127

File: source/javascripts/components/_Workflow.js.erb [hunk 2]
```diff
@@ -78,6 +78,14 @@
 			return !_.contains(this.workflowChain(allWorkflows), workflow);
 		};
 
+		Workflow.prototype.title = function(title) {
+			if (title !== undefined) {
```
[the remark this answers is not in the document]
Comment (AndrasEszes) [reply source/javascripts/components/_Workflow.js.erb:82]: I think wouldn't. The common use-case to remove prop from the yaml is pass `null` to the method.

File: spec/integration/test_bitrise.yml
[the remark this answers is not in the document]
Comment (AndrasEszes) [reply spec/integration/test_bitrise.yml:1]: Nope. I have no clue why is changing sometimes, but when I save the yml through the API it's reformat the whole file.

File: source/javascripts/components/WorkflowConfigPanel/WorkflowConfigPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    onChange(data);
+  });
+
+  return (
+    <FormProvider {...form}>
+      <form onSubmit={handleSubmit} onChange={handleSubmit}>
```
[the remark this answers is not in the document]
Comment (AndrasEszes) [reply source/javascripts/components/WorkflowConfigPanel/WorkflowConfigPanel.tsx:31]: Do you remember? :rofl: 
https://github.com/bitrise-io/bitrise-workflow-editor/blob/master/source/javascripts/components/StepConfigPanel/StepProperties.tsx#L28

**Review by zoltanszabo-bitrise — APPROVED** [review]

**Pull request merged**